core/src/main/java/org/elasticsearch/search/internal/ShardSearchLocalRequest.java

```diff
@@ -232,4 +233,10 @@ public BytesReference cacheKey() throws IOException {
         // we could potentially keep it without copying, but then pay the price of extra unused bytes up to a page
         return out.bytes().copyBytesArray();
     }
+
+    @Override
+    public ShardSearchLocalRequest rewrite(QueryShardContext context) throws IOException {
+        source = source.rewrite(context);
```

Comment:
Not currently, but should I make it rewrite until its unchanged here in case we have a query that rewrites multiple times in the future? Also, the call to `ShardSearchRequest.rewrite()` in `SearchService.createContext()` rewrites the request until its unchanged anyway, I guess this could result is lot of object creation if there are multiple rewrites? should we rewritten until its unchanged at every level?